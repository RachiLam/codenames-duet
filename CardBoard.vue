<template>
  <div class="card-board">
    <card
      v-for="card in cards"
      :key="card.id"
      :card="card"
    />
  </div>
</template>

<script>
import Card from './Card.vue';

export default {
  components: {
    Card,
  },
  props: {
    cards: {
      type: Array,
      required: true,
    },
  },
};
</script>

<style scoped lang="scss">
.card-board {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
</style>
