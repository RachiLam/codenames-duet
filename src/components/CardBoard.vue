<template>
  <div class="card-board">
    <div
      v-for="rowNum in 5"
      :key="`card-row-${rowNum}`"
      class="row"
    >
      <card
        v-for="colNum in 5"
        :key="`card-${getCardId(rowNum, colNum)}`"
        :card-id="getCardId(rowNum, colNum)"
      />
    </div>
  </div>
</template>

<script>
import Card from './Card.vue';

export default {
  components: {
    Card,
  },
  methods: {
    getCardId(rowNum, colNum) {
      return (rowNum - 1) * 5 + (colNum - 1);
    },
  },
};
</script>

<style scoped lang="scss">
@use 'styles/constants';

#card-board {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: center;
  padding: constants.$card-margin;

  .row {
    display: flex;
    flex: 1;
  }
}
</style>
