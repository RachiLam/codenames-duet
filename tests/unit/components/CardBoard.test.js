import CardBoard from 'components/CardBoard.vue';

const { shallowMount } = getMountUtils({
  component: CardBoard,
});

describe('components/CardBoard', () => {
  describe('always', () => {
    it('renders 25 cards in 5 rows', () => {
      const wrapper = shallowMount();
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
