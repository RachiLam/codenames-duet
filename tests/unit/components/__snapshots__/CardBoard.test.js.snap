// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CardBoard always renders 25 cards in 5 rows 1`] = `
<div
  class="card-board"
>
  <div
    class="row"
  >
    <card-stub
      cardid="0"
    />
    <card-stub
      cardid="1"
    />
    <card-stub
      cardid="2"
    />
    <card-stub
      cardid="3"
    />
    <card-stub
      cardid="4"
    />
  </div>
  <div
    class="row"
  >
    <card-stub
      cardid="5"
    />
    <card-stub
      cardid="6"
    />
    <card-stub
      cardid="7"
    />
    <card-stub
      cardid="8"
    />
    <card-stub
      cardid="9"
    />
  </div>
  <div
    class="row"
  >
    <card-stub
      cardid="10"
    />
    <card-stub
      cardid="11"
    />
    <card-stub
      cardid="12"
    />
    <card-stub
      cardid="13"
    />
    <card-stub
      cardid="14"
    />
  </div>
  <div
    class="row"
  >
    <card-stub
      cardid="15"
    />
    <card-stub
      cardid="16"
    />
    <card-stub
      cardid="17"
    />
    <card-stub
      cardid="18"
    />
    <card-stub
      cardid="19"
    />
  </div>
  <div
    class="row"
  >
    <card-stub
      cardid="20"
    />
    <card-stub
      cardid="21"
    />
    <card-stub
      cardid="22"
    />
    <card-stub
      cardid="23"
    />
    <card-stub
      cardid="24"
    />
  </div>
</div>
`;
